### PR TITLE
Lps 161493 | Add functional tests on expose batch actions for headless-delivery and headless-taxonomy-admin

### PR DESCRIPTION
**Pull request by yanan**
Background:
- expose batch actions for headless-delivery
- expose batch actions for headless-taxonomy-admin

Test Plan:
- When with get request and siteId to retrieve taxonomy vocabularies
  Then I can see details of updateBatch, createBatch and deleteBatch inside of action block in response
- Given an asset library is created
When with get request and assetLibraryId to retrieve taxonomy vocabulary
Then I can see details of updateBatch, createBatch and deleteBatch inside of action block in response
- When with get request and siteId to retrieve document folders
Then I can see details of updateBatch, createBatch and deleteBatch inside of action block in response
- Given an asset library is created
When with get request and assetLibraryId to retrieve document folders
Then I can see details of updateBatch, createBatch and deleteBatch inside of action block in response

Ticket:
- https://issues.liferay.com/browse/LPS-161493

### CODE REVIEW

**Comment by yanan** [comment]
ci:test:sf

**Comment by yanan** [comment]
ci:test:relevant

**Comment by yanan** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by yanan** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by yanan** [comment]
ci:test:sf

**Comment by yanan** [comment]
ci:test:relevant

**Comment by yanan** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by yanan** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by yanan** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by yanan** [comment]
Test failures are unrelated to the current pr. In this pr, I've update getDepod and deleteDepot without webdriver by passing the "noSelenium=true" parameter. @magjed4289 @Hijky

**Comment by yanan** [comment]
ci:test:sf

**Comment by yanan** [comment]
ci:test:relevant

**Comment by yanan** [comment]
ci:test:sf

**Comment by yanan** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by yanan** [comment]
ci:test:relevant

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessAdminTaxonomyAPI/ExposeBatchActionForHeadlessAdminTaxonomyAPI.testcase [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+				depotName = "Test Depot Name",
+				noSelenium = "true");
+
+			var response = TaxonomyVocabularyAPI.getTaxonomyVocabularyWithDifferentParameters(assetLibraryId = "${assetLibraryId}");
+		}
+
+		task ("Then I can see details of updateBatch, createBatch and deleteBatch inside of action block in response") {
+			TaxonomyVocabularyAPI.assertProperBatchInActionBlock(
+				assetLibraryId = "${assetLibraryId}",
+				batchActions = "createBatch,updateBatch,deleteBatch",
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessAdminTaxonomyAPI/ExposeBatchActionForHeadlessAdminTaxonomyAPI.testcase:38]: Maybe we can remove the if statements and look directly for PUT, POST, and DELETE? I mean, in the JSON we'll look in the batch section anyways, right?
I would leave the task title as it is, but change the batchActions to the exact type. What do you think about it? 🤔

File: portal-web/test/functional/com/liferay/portalweb/macros/headless/headlessadmintaxonomy/TaxonomyVocabularyAPI.macro [hunk 2]
```diff
@@ -35,7 +35,9 @@ definition {
 			var api = "asset-libraries/${assetLibraryId}/taxonomy-vocabularies";
 		}
 		else {
-			var siteId = JSONGroupAPI._getSiteIdByGroupKey(groupName = "${groupName}");
+			var siteId = JSONGroupAPI._getGroupIdByNameNoSelenium(
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/macros/headless/headlessadmintaxonomy/TaxonomyVocabularyAPI.macro:38]: nice! 👏🏻

**Comment by magjed4289** [comment]
Just started reviewing :)

**Comment by magjed4289** [comment]
ci:test:headless-frontend-functional:nocompile

**Comment by magjed4289** [comment]
ci:forward

**Pull request closed without merging**